platforms/jvm/language-java/src/main/java/org/gradle/api/internal/tasks/compile/DiagnosticToProblemListener.java

```diff
@@ -124,9 +125,10 @@ private static void addLocations(ProblemSpec spec, Diagnostic<? extends JavaFile
 
             // If we know the offsets ...
             // (offset doesn't require line and column to be set, hence the separate check)
-            if (0 < start && start <= end) {
+            // (documentation says that getEndPosition() will be NOPOS iff getPosition() is NOPOS)
+            if (0 < position) {
                 // ... we can report the start and extent
-                spec.offsetInFileLocation(resourceName, start, end - start);
+                spec.offsetInFileLocation(resourceName, start, end - position);
```

Comment:
Not sure if you made this PR before my comment but if you're going to change this to `end - position` then `start` should also be changed to `position`

Reply:
Absolutely right, fixed!